File: app/src/main/java/com/infullmobile/learnmvpgithub/presentation/counter/di/CounterGraph.java
package com.infullmobile.learnmvpgithub.presentation.counter.di;

import android.support.annotation.VisibleForTesting;

import com.infullmobile.learnmvpgithub.presentation.counter.CounterActivity;

import dagger.Component;

public class CounterGraph {
/*
	private final DaggerCounterGraph_CounterComponent.Builder builder;

	public CounterGraph(CounterActivity activity) {
		this.builder = DaggerCounterGraph_CounterComponent
				.builder()
				.applicationComponent(Components.from(activity.getApplication()))
				.counterModule(new CounterModule(activity));
	}

	public void inject(CounterActivity activity) {
		builder.build().inject(activity);
	}

	@VisibleForTesting
	public void setCounterModule(CounterModule module) {
		builder.counterModule(module);
	}

	@CounterScope
	@Component(
			dependencies = ApplicationComponent.class,
			modules = CounterModule.class
	)
	public interface CounterComponent {
		void inject(CounterActivity activity);
	}*/
}

File: app/src/main/java/com/infullmobile/learnmvpgithub/presentation/counter/di/CounterGraph.kt
package com.infullmobile.learnmvpgithub.presentation.counter.di

import android.support.annotation.VisibleForTesting

import com.infullmobile.learnmvpgithub.presentation.application.di.ApplicationComponent
import com.infullmobile.learnmvpgithub.presentation.application.di.Components
import com.infullmobile.learnmvpgithub.presentation.counter.CounterActivity

import dagger.Component

class CounterGraph(activity: CounterActivity) {
    private val builder: DaggerCounterGraph_CounterComponent.Builder

    init {

        this.builder = DaggerCounterGraph_CounterComponent
                .builder()
                .applicationComponent(Components.from(activity.application))
                .counterModule(CounterModule(activity))
    }

    fun inject(activity: CounterActivity) {
        builder.build().inject(activity)
    }

    @VisibleForTesting
    fun setCounterModule(module: CounterModule) {
        builder.counterModule(module)
    }

    @CounterScope
    @Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(CounterModule::class))
    interface CounterComponent {
        fun inject(activity: CounterActivity)
    }
}
